openpnm/utils/Workspace.py

```diff
@@ -53,7 +53,9 @@ def __setitem__(self, name, project):
         if name is None:
             name = self._gen_name()
         if name in self.keys():
-            raise Exception("A project named " + name + " already exists")
+            # To ensures renaming a Project to its current name is allowed
```

Comment:
I still don't understand these 3 lines.  I've never used the __instance__ magic before.  Also I don't understand why if the object is NOT in the project we raise an exception.  I guess if it works, then that is all that matters.